tests/infra/github.py

```diff
@@ -78,8 +78,11 @@ def sanitise_branch_name(branch_name):
         # For simplification, assume that dev tags are only released from main branch
         LOG.debug(f"Considering dev tag {branch_name} as {MAIN_BRANCH_NAME} branch")
         return MAIN_BRANCH_NAME
-    if is_release_tag(branch_name):
-        equivalent_release_branch = f'{BRANCH_RELEASE_PREFIX}{branch_name.split(TAG_RELEASE_PREFIX)[1].split(".")[0]}.x'
+    elif is_release_tag(branch_name):
+        tag_major_version = int(branch_name.split(TAG_RELEASE_PREFIX)[1].split(".")[0])
```

Comment:
Not a huge amount of utility, but probably still more readable than lots of splits and index lookups: https://pypi.org/project/semver/

Reply:
We have a `versionifier.py` that tries to parse to the setuptools version type. eg

```
    v = to_python_version("ccf-1.2.3-dev3-5-deadbeef")
    assert v.release == (1, 2, 3)
    assert v.dev == 3
    assert v.local == "5.deadbeef"
 ```

It's a bit weird (was written purely to generate the version number for `setup.py`), but could be re-used here to lean on existing version parsers?

Reply:
Good spot. There's already a `get_version_from_tag_name()` function a few lines up that extracts the major version more cleanly and that I should have been using..